autoreject/tests/test_autoreject.py

```diff
@@ -71,59 +79,114 @@ def test_autoreject():
+            ar.local_reject_.n_interpolate[ch_type])
+        assert_equal(
+            ar.consensus_perc_[ch_type],
+            ar.local_reject_.consensus_perc[ch_type])
+
+    # test complementarity of goods and bads
+    assert_array_equal(
+        np.sort(np.r_[bad_epochs_idx, good_epochs_idx]),
+        np.arange(len(epochs_fit)))
+
+    # test that transform does not change state of ar
+    epochs_fit.fit_ = True
+    epochs_clean = ar.transform(epochs_fit)  # apply same data
+    assert_array_equal(fix_log, ar.fix_log)
+    assert_array_equal(bad_epochs_idx, ar.local_reject_.bad_epochs_idx_)
+    assert_array_equal(good_epochs_idx, ar.local_reject_.good_epochs_idx_)
+
+    epochs_new_clean = ar.transform(epochs_new)  # apply to new data
+    assert_array_equal(fix_log, ar.fix_log)
+    assert_array_equal(bad_epochs_idx, ar.local_reject_.bad_epochs_idx_)
+    assert_array_equal(good_epochs_idx, ar.local_reject_.good_epochs_idx_)
+
+    is_same = epochs_new_clean.get_data() == epochs_new.get_data()
+    if not np.isscalar(is_same):
+        is_same = np.isscalar(is_same)
+    assert_true(not is_same)
+
+    assert_equal(epochs_clean.ch_names, epochs_fit.ch_names)
+    # Now we test that the .bad_segments has the shape
+    # of n_trials, n_sensors, such that n_sensors is the
+    # the full number sensors, before picking. We, hence,
+    # expect nothing to be rejected outside of our picks
+    # but rejections can occur inside our picks.
+    assert_equal(ar.bad_segments.shape[1], len(epochs_fit.ch_names))
+    assert_true(np.any(ar.bad_segments[:, picks]))
+    non_picks = np.ones(len(epochs_fit.ch_names), dtype=bool)
+    non_picks[picks] = False
+    assert_true(not np.any(ar.bad_segments[:, non_picks]))
+
+    assert_true(isinstance(ar.threshes_, dict))
+    assert_true(len(ar.picks) == len(picks))
+    assert_true(len(ar.threshes_.keys()) == len(ar.picks))
+    pick_eog = mne.pick_types(epochs.info, meg=False, eeg=False, eog=True)
+    assert_true(epochs.ch_names[pick_eog] not in ar.threshes_.keys())
+    assert_raises(
+        IndexError, ar.transform,
+        epochs.copy().pick_channels(
+            [epochs.ch_names[pp] for pp in picks[:3]]))
+
+    epochs.load_data()
+    assert_raises(ValueError, compute_thresholds, epochs, 'dfdfdf')
+    index, ch_names = zip(*[(ii, epochs_fit.ch_names[pp])
+                          for ii, pp in enumerate(picks)])
+    threshes_a = compute_thresholds(
+        epochs_fit, picks=picks, method='random_search')
+    assert_equal(set(threshes_a.keys()), set(ch_names))
+    threshes_b = compute_thresholds(
+        epochs_fit, picks=picks, method='bayesian_optimization')
+    assert_equal(set(threshes_b.keys()), set(ch_names))
+
+    # arrays are really different but quite similar.
+    # XXX this actually varies quite substantially
+    # assert_true(np.corrcoef(
+    #                 list(threshes_a.values()),
+    #                 list(threshes_b.values()))[0, 1] ** 2 > 0.8)
```

Comment:
let's remove this? Random search didn't work very well if I remember correctly. That's why bayesian optimization is the default